app/services/notification/process.rb

```diff
@@ -0,0 +1,99 @@
+require 'enumerate_it'
+
+class Notification::Process
+  class TransactionStatus < ::EnumerateIt::Base
+    associate_values(
+      authorized:     1,
+      started:        2,
+      printed_boleto: 3,
+      finished:       4,
+      canceled:       5,
+      process:        6,
+      written_back:   7,
+      refunded:       9
+    )
+  end
+
+  def initialize(params)
+    @params  = params
+  end
+
+  def self.call(params)
+    new(params).call
+  end
+
+  def call
+    process_moip_message unless payment.nil?
+  end
+
+  private
+  attr_accessor :params
+
+  def payment
+    @payment ||= find_payment
+  end
+
+  def find_payment
+    payment = PaymentEngines.find_payment(key: params[:id_transacao])
+    CatarseMoip::PaymentEngineDecorator.decorate(payment) unless payment.nil?
+  end
+
+  def process_moip_message
+    payment.transaction do
+      notification
+      process_payment
+    end
+  end
+
+  def notification
+    @notification ||= CatarseMoip::NotificationDecorator.decorate(payment.notify_payment(extra_data: parse_params))
+  end
+
+  def process_payment
+    return if !update_payment_id? && invalid_payment?
+
+    notify(params[:status_pagamento].to_i)
+  end
+
+  def update_payment_id?
+    return if parsed_payment_id > params[:cod_moip].to_i
+
+    payment.update(payment_id: params[:cod_moip])
+  end
+
+  def invalid_payment?
+    return if params[:valor].nil?
+
+    (params[:valor].to_i/100.0) < payment.value
+  end
+
+  def notify(status_payment)
```

Comment:
This method is doing more than just notifying. Some cases you're notifying some you're processing the payment. I'd suggest you to rename this method or separate these two different logic in different methods.